src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts

```diff
@@ -20,6 +22,24 @@ export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWiza
             if (!result) {
                 context.version = await promptForFuncVersion(context);
             } else {
+                const endofLifeDate = result.minorVersion.stackSettings.linuxRuntimeSettings?.endOfLifeDate;
+                const sixMonthsFromNow = new Date();
+                sixMonthsFromNow.setMonth(sixMonthsFromNow.getMonth() + 6);
+                const settingKey: string = 'endOfLifeWarning';
+                if (getWorkspaceSetting<boolean>('endOfLifeWarning')) {
+                    if (endofLifeDate) {
+                        if (compareDates(endofLifeDate, sixMonthsFromNow)) {
+                            const message = localize('endOfLife', "The chosen runtime stack has an end of support deadline coming up. After the deadline, function apps can be created and deployed, and existing apps continue to run. However, your apps won't be eligible for new features, security patches, performance optimizations, and support until you upgrade them");
+                            const result: MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.learnMore, DialogResponses.dontWarnAgain);
```

Comment:
Ideally selecting "Back" here would take the user back to the stack step, instead of back to the name step.